LEGO1/mxbitmap.cpp

```diff
@@ -0,0 +1,77 @@
+#include "mxbitmap.h"
+#include "mxbool.h"
+#include "mxpalette.h"
+#include "mxresult.h"
+
+// OFFSET: LEGO1 0x100bc980
+MxBitmap::MxBitmap()
+{
+  this->m_info = NULL;
+  this->m_bmiHeader = NULL;
+  this->m_paletteData = NULL;
+  this->m_data = NULL;
+  this->m_unk18 = MX_FALSE;
+  this->m_palette = NULL;
+}
+
+// OFFSET: LEGO1 0x100bca10
+MxBitmap::~MxBitmap()
+{
+  if (this->m_info != NULL) {
+    delete m_info;
+  }
+  if (this->m_data != NULL) {
+    delete m_data;
+  }
+  if (this->m_palette != NULL) {
+    delete m_palette;  
+  }
+}
+
+// OFFSET: LEGO1 0x100bd1c0
+MxPalette *MxBitmap::CreatePalette()
+{
+  // FIXME: This function needs MxPalette to be completed. Also INFERRING usage of MxBool
+  MxPalette *pal = NULL;
+  MxPalette *ppal;
+  MxBool success = MX_FALSE;
+
+  if(this->m_unk18 == MX_FALSE) {
+    // ppal = MxPalette::FromBitmapPalette(this->m_paletteData);
```

Comment:
This is really a constructor with `new` because `operator_new` is called in this method.
```suggestion
    // ppal = new MxPalett(this->m_paletteData); // FIXME
```